### PR TITLE
Properly support sending InputStream in REST Client

### PR DESCRIPTION
**Pull request by geoand**
This means **not** keeping the entire content of the stream in memory

### CODE REVIEW

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InputStreamReadStream.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            }
+            if (amount == -1) {
+                p.complete();
+            } else {
+                p.complete(
+                        Buffer.buffer(VertxByteBufAllocator.DEFAULT.heapBuffer(amount, Integer.MAX_VALUE).writeBytes(bytes, 0,
+                                amount)));
+            }
+        });
+        fut.onComplete(ar -> {
```
Comment (cescoffier) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InputStreamReadStream.java:117]: Lambdas :-)

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java [hunk 2]
```diff
@@ -165,6 +167,12 @@ public void handle(AsyncResult<AsyncFile> openedAsyncFile) {
                                             attachSentHandlers(sent, httpClientRequest, requestContext);
                                         }
                                     });
+                } else if (requestContext.isInputStreamUpload()) {
+                    Future<HttpClientResponse> sent = httpClientRequest.send(
+                            new InputStreamReadStream(
+                                    Vertx.currentContext().owner(), (InputStream) requestContext.getEntity().getEntity(),
```
Comment (cescoffier) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java:173]: Not sure - I believe you want the connection one.

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InputStreamReadStream.java (hunk 1 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/InputStreamReadStream.java:117]: Ah... I'll fix it

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java (hunk 2 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java:173]: We have already been using the same thing in Multipart handling without issue

Comment (cescoffier) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java:177]: Ok.

**Review by cescoffier — APPROVED** [review]

**Pull request merged**